src/js/overlay.js

```diff
@@ -225,16 +219,33 @@ Overlay.prototype.show = function() {
 
 	this.closeOnExternalClick = this.closeOnExternalClick.bind(this);
 	this.delegates.doc.on('click', 'body', this.closeOnExternalClick);
+	this.delegates.doc.on('touchstart', 'body', this.closeOnExternalClick);
```

Comment:
Right, some taps can be start of a swipe or system-wide multitasking gesture. In that case `touchend` is better, or ideally note target node from `touchstart` and ensure it's the same in `touchend`.

If you `preventDefault()` in touch handlers you won't get mouse events. In that case it may not be an issue — I assume attempt to close the overlay for the second time just won't do anything.